### PR TITLE
no longer use the internal `TestFailure` class

### PR DESCRIPTION
**Pull request by xabbuh**
| Q             | A
| ------------- | ---
| Branch?       | 7.2
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Issues        | 
| License       | MIT

The class was removed in PHPUnit 10.

### CODE REVIEW

File: src/Symfony/Component/HttpFoundation/Tests/Test/Constraint/ResponseIsRedirectedTest.php [hunk 1]
```diff
@@ -29,9 +28,8 @@ public function testConstraint()
         try {
             $constraint->evaluate(new Response('Body content'));
         } catch (ExpectationFailedException $e) {
-            $exceptionMessage = TestFailure::exceptionToString($e);
-            $this->assertStringContainsString("Failed asserting that the Response is redirected.\nHTTP/1.0 200 OK", $exceptionMessage);
-            $this->assertStringContainsString('Body content', $exceptionMessage);
+            $this->assertStringContainsString("Failed asserting that the Response is redirected.\nHTTP/1.0 200 OK", $e->getMessage());
+            $this->assertStringContainsString('Body content', $e->getMessage());
```
Comment (derrabus) [inline src/Symfony/Component/HttpFoundation/Tests/Test/Constraint/ResponseIsRedirectedTest.php]: Can't we work with `expectException()` here too?

File: src/Symfony/Component/HttpFoundation/Tests/Test/Constraint/ResponseStatusCodeSameTest.php [hunk 2]
```diff
@@ -48,9 +46,8 @@ public function testReducedVerbosity()
         try {
             $constraint->evaluate(new Response('Response body', 404));
         } catch (ExpectationFailedException $e) {
-            $exceptionMessage = TestFailure::exceptionToString($e);
-            $this->assertStringContainsString("Failed asserting that the Response status code is 200.\nHTTP/1.0 404 Not Found", TestFailure::exceptionToString($e));
-            $this->assertStringNotContainsString('Response body', $exceptionMessage);
+            $this->assertStringContainsString("Failed asserting that the Response status code is 200.\nHTTP/1.0 404 Not Found", $e->getMessage());
+            $this->assertStringNotContainsString('Response body', $e->getMessage());
```
Comment (derrabus) [inline src/Symfony/Component/HttpFoundation/Tests/Test/Constraint/ResponseStatusCodeSameTest.php:45]: You didn't cause this, but this test looks flawed. It passes if no exception is raised and I don't believe that this is intended.

**Review by derrabus — COMMENTED** [review]
Great work, thank you!

File: src/Symfony/Component/HttpFoundation/Tests/Test/Constraint/ResponseStatusCodeSameTest.php (hunk 2 above)
Comment (xabbuh) [reply src/Symfony/Component/HttpFoundation/Tests/Test/Constraint/ResponseStatusCodeSameTest.php:45]: see #58178

File: src/Symfony/Component/HttpFoundation/Tests/Test/Constraint/ResponseIsRedirectedTest.php (hunk 1 above)
Comment (xabbuh) [reply src/Symfony/Component/HttpFoundation/Tests/Test/Constraint/ResponseIsRedirectedTest.php]: yes indeed, I updated some more tests accordingly

**Review by derrabus — APPROVED** [review]

**Review by stof — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @xabbuh.

**Pull request merged**